shell/platform/android/io/flutter/plugin/editing/FlutterTextUtils.java

```diff
@@ -0,0 +1,184 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugin.editing;
+
+import io.flutter.embedding.engine.FlutterJNI;
+
+class FlutterTextUtils {
+  public static final int LINE_FEED = 0x0A;
+  public static final int CARRIAGE_RETURN = 0x0D;
+  public static final int COMBINING_ENCLOSING_KEYCAP = 0x20E3;
+  public static final int CANCEL_TAG = 0xE007F;
```

Comment:
I believe the Unicode spec refers to this as `TAG_END`, should probably stick to same naming scheme

Reply:
http://unicode.org/reports/tr51/#def_emoji_tag_sequence

Reply:
Jk, it looks like they do call this `CANCEL_TAG`, this is fine.

Reply:
I think it's better to follow the Android guideline because we are using it as the reference.